node/blockAPI/shardBlock.go

```diff
@@ -67,6 +67,21 @@ func (sbp *shardAPIBlockProcessor) GetBlockByHash(hash []byte, withTxs bool) (*a
 	return sbp.computeStatusAndPutInBlock(blockAPI, storerUnit)
 }
 
+// GetBlockByRound will return a shard APIBlock by round
+func (sbp *shardAPIBlockProcessor) GetBlockByRound(round uint64, withTxs bool) (*api.Block, error) {
```

Comment:
duplicated code with metaBlock.go. We usually fix these by creating a base struct with the common functionality and extend it in all implementations.

Reply:
Yes. I was aware about it, but couldn't use the same implementation for both meta and shard blocks, because the returned value is given by: `GetBlockByNonce(nonce, withTxs)` -> which has different implementations for meta/shard block.
However, I've extracted some common code in a separate function, which is now part of `baseBlocks`:

```go
// GetBlockByRound will return a shard APIBlock by round
func (bap *baseAPIBlockProcessor) getNonceParamByRound(round uint64, withTxs bool) (uint64, error) {
	roundToByteSlice := bap.uint64ByteSliceConverter.ToByteSlice(round)
	nonceBytes, err := bap.store.Get(dataRetriever.RoundNonceUnit, roundToByteSlice)
	if err != nil {
		return 0, err
	}

	nonce, err := bap.uint64ByteSliceConverter.ToUint64(nonceBytes)
	if err != nil {
		return 0 , err
	}
	return nonce, nil
}
```

Reply:
The above comment is now obsolete, since the block's hash is now saved by round instead of saving block's round by nonce